### ISSUE TITLE
page 2 blended with page 1 on output via append

### ISSUE THREAD

**Issue by kimberv** [py-pdf/pypdf#1595]
I need to read in a fillable form pdf, fill it out, then generate result pdf. this is largely working, except that page 2 of the pdf (which contains only text) doesn't get properly appended. It's text contents are laid over the top of a copy of page 1. 

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
macOS-11.6.5-x86_64-i386-64bit

$ python -c "import pypdf;print(pypdf.__version__)"
3.3.0
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from pypdf import PdfReader, PdfWriter

reader = PdfReader('cms1500.pdf')
writer = PdfWriter()
writer.append(reader)

with open('copy.pdf', "wb") as output_stream:
    writer.write(output_stream)
```

Pdf source is 
[form-cms1500.pdf](https://github.com/py-pdf/pypdf/files/10564431/form-cms1500.pdf) - a standard pdf in healthcare. There should be no problem with you using it in test suites.

## Results

Outcome from using `writer.append`:
<img width="713" alt="Screen Shot 2023-02-01 at 10 21 22 PM" src="https://user-images.githubusercontent.com/191307/216223762-0e02d674-3127-4663-b830-5c787badba53.png">

Outcome from using `writer.append(reader, [1])`
<img width="643" alt="Screen Shot 2023-02-01 at 10 22 48 PM" src="https://user-images.githubusercontent.com/191307/216223842-f5e8e0c5-e034-4aef-9313-34311a52dedb.png">

NOTE: Outcome from using `writer.clone_document_from_reader` does produce expected output **BUT i can't use it as it causes other problems (namely after clone writer.pages items appear to be DictionaryObject and not PageObject)**:
<img width="611" alt="Screen Shot 2023-02-01 at 10 24 26 PM" src="https://user-images.githubusercontent.com/191307/216223941-407dd23a-7e3f-4219-8d10-e4f6f4503025.png">

**Comment by kimberv** [comment]
I'd love any insight on the above issue or if you need any additional information.

**Comment by pubpub-zz** [comment]
I've started to have a look and I'm very confused : the page content shown no different when copied into the new PDF. 
The big difference when using `clone_document_from_reader()` you are copying some fields/object from "/Root".
I have not been able to identify yet where's the fields modifying the behavior

Meanwhile, I've prepared a PR for returning `PageObject` when using `clone_document_from_reader()`, can you check it

**Comment by pubpub-zz** [comment]
@kimberv, you looked in a hurry... Can you provide update?

**Comment by kimberv** [comment]
I can confirm that this change allows me to use `writer.clone_document_from_reader(reader)` as expected, and the resulting pdf has a "clean" page 2! I look forward to this patch being part of a release! (to be clear to any curious reader though - `append(reader)` still has the above problem for this pdf)

**Comment by pubpub-zz** [comment]
As sais, the difference of behavior between page1 and 2 is due to the "/OCProperties" within "/Root". There is no indirect object making link with the pages : `Append()` will not be able to copy these data

**Comment by MartinThoma** [comment]
The fix was just merged and will be part of the `pypdf>3.4.1` release this weekend (26.02.2023)
